src/librustc_typeck/astconv.rs

```diff
@@ -102,6 +102,7 @@ enum GenericArgPosition {
 /// Dummy type used for the `Self` of a `TraitRef` created for converting
 /// a trait object, and which gets removed in `ExistentialTraitRef`.
 /// This type must not appear anywhere in other converted types.
+#[cfg_attr(not(stage0), allow(usage_of_ty_tykind))]
```

Comment:
This constant is used in comparisons. Should this lint ignore constant definitions completely?

Reply:
I think we should just add it to the `tcx.types` list: https://github.com/rust-lang/rust/blob/9ebf47851a357faa4cd97f4b1dc7835f6376e639/src/librustc/ty/context.rs#L201